readthedocs/proxito/middleware.py

```diff
@@ -215,10 +211,11 @@ def _set_request_attributes(self, request, unresolved_domain):
                     project_slug=project.slug,
                 )
                 request.canonicalize = constants.REDIRECT_SUBPROJECT_MAIN_DOMAIN
-        else:
-            raise NotImplementedError
 
     def process_request(self, request):  # noqa
+        # Initialize our custom request attributes.
+        request.unresolved_domain = None
```

Comment:
It felt better to just initialize this to `None`, rather than using `getattr` everywhere.

Reply:
Here is the answer to my question 😄